### PR TITLE
Net 137p - Pull requests

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
update pull requests for relay refactor
## Provide testing steps
see NET-137p

### CODE REVIEW

File: logic/peers.go [hunk 1]
```diff
@@ -679,3 +682,187 @@ func filterNodeMapForClientACLs(publicKey, network string, nodePeerMap map[strin
 	}
 	return nodePeerMap
 }
+
+func GetPeerUpdate(host *models.Host) []wgtypes.PeerConfig {
+	peerUpdate := []wgtypes.PeerConfig{}
+	for _, nodeStr := range host.Nodes {
+		node, err := GetNodeByID(uuid.MustParse(nodeStr).String())
```
Comment (abhishek9686) [inline logic/peers.go:689]: `nodeStr` is a string, why parse it again?

File: logic/relay.go [hunk 2]
```diff
@@ -135,16 +136,116 @@ func getRelayedAddresses(id string) []net.IPNet {
...
+		logger.Log(0, "GetPeerUpdateForRelayed called for non-relayed node ", client.Host.Name)
+		return []wgtypes.PeerConfig{}
+	}
+	relayNode, err := GetNodeByID(client.Node.RelayedBy)
+	if err != nil {
+		logger.Log(0, "error retrieving relay node", err.Error())
+		return []wgtypes.PeerConfig{}
+	}
+	relay := models.Client{
+		Host: *GetHostByNodeID(relayNode.ID.String()),
```
Comment (abhishek9686) [inline logic/relay.go:162]: use node.HostID to get Host

File: logic/peers.go [hunk 3]
```diff
@@ -679,3 +682,187 @@ func filterNodeMapForClientACLs(publicKey, network string, nodePeerMap map[strin
...
+				cidr.IP = ip
+				ingressIPs = append(ingressIPs, *cidr)
+			}
+		}
+	}
+	return ingressIPs
+}
+
+// GetPeerUpdateForRelay - returns the peer update for a relay node
+func GetPeerUpdateForRelay(client *models.Client, peers *[]models.Client) []wgtypes.PeerConfig {
```
Comment (abhishek9686) [inline logic/peers.go:845]: why pointer to slice?

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: logic/peers.go [hunk 4]
```diff
@@ -193,7 +194,11 @@ func GetPeerUpdateForHost(ctx context.Context, network string, host *models.Host
 					}
 				}
 				if peer.IsEgressGateway {
-					allowedips = append(allowedips, getEgressIPs(&node, &peer)...)
+					allowedips = append(allowedips, getEgressIPs(
+						&models.Client{
+							Host: *GetHostByNodeID(peer.ID.String()),
+							Node: peer,
```
[the remark this answers is not in the document]
Comment (mattkasun) [reply logic/peers.go:199]: single return vs double return

**Review by TobiaszCudnik — APPROVED** [review]

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**